sympy/solvers/inequalities.py

```diff
@@ -982,3 +985,487 @@ def reduce_inequalities(inequalities, symbols=[]):
+    eqns = []
+
+    for rel in constraints:
+        if not isinstance(rel, (Relational, AppliedBinaryRelation)):
+            raise TypeError(f"{rel} is not relational.")
+        if type(rel) in [Lt, Gt] or isinstance(rel, AppliedBinaryRelation) and rel.function in [Q.lt, Q.gt]:
+            raise TypeError("Strict inequalities are not allowed in linear programming.")
+        if type(rel) == Ne or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.ne:
+            raise TypeError("'not equal to' is not allowed in linear programming.")
+
+        if type(rel) == Le or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.le:
+            eqns.append(rel.lhs - rel.rhs)
+            standard_constraints.append(rel.lhs <= rel.rhs)
+        elif type(rel) == Ge or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.ge:
+            eqns.append(rel.rhs - rel.lhs)
+            standard_constraints.append(-rel.lhs <= -rel.rhs)
+        elif type(rel) == Eq or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.eq:
+            # x = 3 can be represented as x <= 3 and x >= 3
+            eqns.append(rel.lhs - rel.rhs)
+            standard_constraints.append(rel.lhs <= rel.rhs)
+            eqns.append(rel.rhs - rel.lhs)
+            standard_constraints.append(-rel.lhs <= -rel.rhs)
+        else:
+            raise TypeError(f"Unrecognized relation: {rel}")
+
+    A, B = linear_eq_to_matrix(eqns, *variables)
+    C = linear_eq_to_matrix(objective, *variables)[0] # constant terms can be safely ignored here
+
+    return A, B, C, standard_constraints
+
+
+def linprog_maximize_from_equations(constraints, objective, variables):
+    """
+    A function to maximize a linear objective function subject to linear
+    constraints with the simplex method. All coefficients must be Rational or
+    floats.
+
+    Parameters
+    ==========
+
+    constraints : list of linear inequalities
+        Stict inequalities (>, <) and not equals are not allowed.
+
+    objective : a linear function to maximize
+        If you want to minimize a function, f, simply make the objective
+        function -f.
+
+    variables : a list of variables included in constraints and objective
+
+    Returns
+    =======
+
+    Returns `(optimum, argmax, argmax_dual)`:
+
+    optimum : Rational or Float
+
+    argmax : dictionary of variables to Rationals or Floats
+
+    argmax_dual : dictionary of constraints to Rationals or Floats
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.solvers.inequalities import linprog_maximize_from_equations
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmax_dual  = linprog_maximize_from_equations([r1,r2,r3], x+y+5*z, [x, y, z])
+    >>> optimum
+    11/3
+    >>> argmax
+    {x: 0, y: 1/3, z: 2/3}
+    >>> argmax_dual
+    {-x - 3*z <= -2: 2/3, y + 2*z <= 3: 0, 2*x + y + 7*z <= 5: 1}
```

Comment:
What does `-x - 3*z <= -2: 2/3` mean?

Reply:
Each constraint is associated with one of the dual variables. So that's just a mapping of each constraint to the values that minimize the dual programing problem. I'm not sure how useful this is. It might be better to return [2/3, 0, 1] instead.

Reply:
The formatting of that last thing you said got messed up: 
> It might be better to return the dual system's constraints, objective function and variables. But I don't use this at all. Does anyone else have an idea of whether/how to return the dual information? I think mapping it to an inequality as done here is not so transparent as to its meaning.

Reply:
See [this](https://or.stackexchange.com/questions/4177/physical-interpretation-of-a-dual-of-an-lp#:~:text=The%20dual%20variables%20represent%20the,in%20each%20primal%20constraint%20limit.) stack exchange post for more information on what the dual variables represent. 

My understanding of the post is that the value of the dual variable associated with each constraint tell you how sensitive the optimum value is to changes in that constraint.

Reply:
Thanks for the reference re dual. I would also highly recommend the Ferguson discussion on page 14 of the pdf. I recall now going through some of the questions that he poses.

I would like to see the non-homogeneous form be recognized. Although this is trivial to see the maximization of `2x+y+3` is the same as the maximization of `2x+y`, I don't thing there is a good reason to not allow the natural expression of the function of interest. BUT that constant can be extracted and handled at the `linprog...` level, not at the `simplex` level: so simplex can always not have to worry about a non-zero `D` (in nomenclature of the clip I gave). And I think `simplex` should just take matrix `M` instead of A, B, C. What do you think?

Reply:
I'm not too familiar with either linear programming or the simplex method. Is it unambiguous to say "here is a matrix M, please do simplex on it to solve a linear programming problem"?

If that is unambiguous then we can have a function `simplex(M)` but then it also suggests that we could have a matrix method `M.simplex()`.

Reply:
> BUT that constant can be extracted and handled at the linprog... level

Right now the constant is just completely ignored in `linprog_from_equations`, which I just realized is problematic because the constant term should affect the final optimum. I agree that `simplex` should be able to take a non-zero D.

> And I think simplex should just take matrix M instead of A, B, C. What do you think?

In earlier versions of `simplex` it just took `M`. I changed it to avoid duplicate code in the linear programming functions and find feasible. I think going back to just using `M` is also reasonable.

Reply:
>  we could have a matrix method M.simplex()

We could. I don't think it would be very useful outside of the context of the functions inside this PR though.